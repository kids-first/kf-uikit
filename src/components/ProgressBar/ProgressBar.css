.ProgressBar {
    @apply
        w-full
        font-light;
}

.ProgressBar--stat {
    @apply
        float-right;
}

.ProgressBar--bar {
    @apply
        inline-block
        max-w-full
        h-4
        rounded-lg
        leading-none;
}

.ProgressBar--bar-well {
    @apply
        w-full
        bg-grey-light;
}
