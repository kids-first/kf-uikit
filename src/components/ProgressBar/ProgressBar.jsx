import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';

/**
 * A linear progress bar of varying color.
 */
const ProgressBar = ({ total, current, color, className }) => {

  const baseClass = classes(
      'ProgressBar',
      className
  );
  const barClass = classes(
    'ProgressBar--bar',
    `ProgressBar--bar-${color}`,
    className
  );

  // Don't allow progress bar to squish too small
  const progress = Math.max(current/total*100, 1.0);

  return (
    <div className={baseClass}>
      <span className='ProgressBar--title'>Number of Files</span>
      <span className='ProgressBar--stat'>
        {current}
        /
        {total}
      </span>
      <div className='ProgressBar--bar ProgressBar--bar-well'>
        <span className={barClass} style={{width: `${progress}%`}} />
      </div>
    </div>
  );
}

ProgressBar.propTypes = {
  /** The number at which completeness occurs */
  total: propTypes.number,
  /** The current progress */
  current: propTypes.number,
  /** The color of the button. */
  color: propTypes.string,
  /** Any additional classes to be applied to the button */
  className: propTypes.string,
};

ProgressBar.defaultProps = {
  total: 100.0,
  current: 0.0,
  color: 'primary',
  className: null,
};

export default ProgressBar;
