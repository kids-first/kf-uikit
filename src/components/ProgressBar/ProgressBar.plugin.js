/* eslint-disable import/no-extraneous-dependencies */
const _ = require('lodash')
const defaultConfig = require('tailwindcss/defaultConfig')()

function defaultOptions() {
  return {
    colors: {
      white: {
        background: defaultConfig.colors.white,
      },
      black: {
        background: defaultConfig.colors.black,
      },
    },
  }
}

module.exports = (opts) => {
  const options = _.isFunction(opts)
   ? opts(defaultOptions())
   : _.defaults(opts, defaultOptions())

  return ({ addComponents, e }) => {
    addComponents([
      ..._.map(options.colors, (colorOptions, name) => ({
        [`.ProgressBar--bar-${e(name)}`]: {
          display: 'inline-block',
          backgroundColor: colorOptions.background,
          height: '100%',
        }
      }))
    ])
  }
}
