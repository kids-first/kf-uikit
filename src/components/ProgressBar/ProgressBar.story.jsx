/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, number } from '@storybook/addon-knobs';
import ProgressBar from './ProgressBar';

const stories = storiesOf('ProgressBars', module);

stories.add('Simple ProgressBar', () => (
  <ProgressBar
    color={select('Color', ['primary', 'secondary', 'tertiary', 'info'], 'primary')}
    current={number('Current', 30)}
    total={number('Total', 100)}
  >
    Click Me
  </ProgressBar>
  ),
  {
		info: {
      text: `
        The progress bar should be used to shown a percentage complete or
        available.
        `
    },
  }
)
