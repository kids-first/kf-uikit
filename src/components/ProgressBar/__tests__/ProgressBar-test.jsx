import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../ProgressBar';

['primary', 'secondary', 'tertiary'].map((c) => {
  [0, 10, 10000].map((current) => {
    [10, 100, 100000].map((total) => {
      it(`${current}/${total} ${c} progress bar renders correctly`, () => {
          const tree = renderer.create(
                  <ProgressBar
                    color={c}
                    current={current}
                    total={total}
                  />
                  ).toJSON();
          expect(tree).toMatchSnapshot();
      });
    });
  });
});
