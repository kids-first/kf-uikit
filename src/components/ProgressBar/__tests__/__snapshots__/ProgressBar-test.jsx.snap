// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`0/10 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/10 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/10 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100000 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100000 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`0/100000 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    0
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/10 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/10 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/10 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100000 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100000 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`10/100000 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "1%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/10 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "100000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/10 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "100000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/10 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    10
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "100000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "10000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "10000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "10000%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100000 primary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-primary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100000 secondary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-secondary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`10000/100000 tertiary progress bar renders correctly 1`] = `
<div
  className="ProgressBar"
>
  <span
    className="ProgressBar--title"
  >
    Number of Files
  </span>
  <span
    className="ProgressBar--stat"
  >
    10000
    /
    100000
  </span>
  <div
    className="ProgressBar--bar ProgressBar--bar-well"
  >
    <span
      className="ProgressBar--bar ProgressBar--bar-tertiary"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;
